Description: Fix FTBFS on hurd-i386, PATH_MAX and MAXSYMLINKS are not defined
 on Hurd. maxsymlinks() from [0] by Justus Winter
 [0] http://anonscm.debian.org/gitweb/?p=collab-maint/sysvinit;a=commitdiff;h=b2db6477ee38490a593b14d5ae20f1bef86e65d2
Author: Gabriele Giacone <1o5g4r8o@gmail.com>, Justus Winter <4winter@informatik.uni-hamburg.de>

--- a/src/common/canonicalize.c
+++ b/src/common/canonicalize.c
@@ -32,6 +32,24 @@
 #include <stddef.h>
 #include <config.h>
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
+/*
+ * Get the maximal number of symlinks to follow.
+ */
+
+static int maxsymlinks(void)
+{
+	int v = sysconf(_SC_SYMLOOP_MAX);
+#ifdef MAXSYMLINKS
+	if (v == -1)
+		return MAXSYMLINKS;
+#endif
+	return v;
+}
+
 /*
  * Return the canonical absolute name of file NAME.  A canonical name does not
  * contain any `.', `..' components nor any repeated path separators ('/') or
@@ -159,7 +177,7 @@ char *vde_realpath(const char *name, cha
 				{
 					size_t len;
 
-					if (++num_links > MAXSYMLINKS)
+					if (++num_links > maxsymlinks())
 					{
 						errno = ELOOP;
 						goto abort;
--- a/include/vdeplugin.h
+++ b/include/vdeplugin.h
@@ -124,4 +124,8 @@ void qtime_csexit();
 unsigned int qtimer_add(time_t period,int times,void (*call)(),void *arg);
 void qtimer_del(unsigned int n);
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 #endif
--- a/include/vde.h
+++ b/include/vde.h
@@ -24,4 +24,8 @@
  */
 #define VDE_PQ
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 #endif
--- a/src/vde_plug2tap.c
+++ b/src/vde_plug2tap.c
@@ -265,6 +265,7 @@ static ssize_t vde_plug2tap_recv(void *o
 
 int main(int argc, char **argv)
 {
+#ifndef VDE_GNU
 	static char *sockname=NULL;
 	static char *tapname=NULL;
 	int daemonize=0;
@@ -417,4 +418,5 @@ int main(int argc, char **argv)
 
 	}
 	return(0);
+#endif
 }
--- a/include/config.h.in
+++ b/include/config.h.in
@@ -312,6 +312,9 @@
 /* If defined, this is a kFreeBSD system */
 #undef VDE_KFREEBSD
 
+/* If defined, this is a Hurd system */
+#undef VDE_GNU
+
 /* If defined, this is a Linux system */
 #undef VDE_LINUX
 
--- a/configure.ac
+++ b/configure.ac
@@ -115,9 +115,12 @@ case "$build_os" in
   freebsd*)
     AC_DEFINE([VDE_FREEBSD], 1, [If defined, this is a FreeBSD system])
   ;;
+  gnu*)
+    AC_DEFINE([VDE_GNU], 1, [If defined, this is a Hurd system])
+  ;;
   *)
-    AC_MSG_ERROR([Unsupported OS: $build_os. At the moment, only Linux, Darwin
-and FreeBSD are supported. Contributions are appreciated! :-)])
+    AC_MSG_ERROR([Unsupported OS: $build_os. At the moment, only Linux, Darwin,
+FreeBSD and Hurd are supported. Contributions are appreciated! :-)])
   ;;
 esac
 
